spark/src/main/java/org/apache/iceberg/Actions.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import org.apache.spark.sql.SparkSession;
+
+public class Actions {
+
+  private SparkSession spark;
+  private Table table;
+
+  private Actions(SparkSession spark, Table table) {
+    this.spark = spark;
+    this.table = table;
+  }
+
+  public static Actions forTable(SparkSession spark, Table table) {
+    return new Actions(spark, table);
+  }
+
+  public static Actions forTable(Table table) {
+    return new Actions(SparkSession.active(), table);
+  }
+
+  public RemoveOrphanFilesAction removeOrphanFiles() {
```

Comment:
We should add Javadoc here that explains what happens. That should note that the table location needs to be accessible for listing via Hadoop FileSystem, but the table's FileIO will be used to delete.

Reply:
Will add docs in the next pass. If we decide to extend `FileIO` with list operations too, we should migrate to that here.

Reply:
Hm. I've been avoiding a list operation in `FileIO`. Maybe we could add it as an optional trait? I just don't want that interface to get huge and complicated.

Reply:
I would not worry about that for now. We will need to expose some sort of `FileStatus` too if we decide to support list operations. Let's first have that use case when the location cannot be listed by Hadoop `FileSystem` and requires a custom `FileIO` implementation.

Reply:
I agree